<?xml version="1.0"?>
<robot name="gripper">
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.045 0.005 0.01"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
      <origin xyz="0 0 0" rpy="0 0 0"/>
    </visual>
  </link>

  <link name="left_finger">
    <visual>
      <geometry>
        <box size="0.055 0.005 0.01"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="0.0275 0 0"/>
    </visual>
  </link>

  <link name="right_upper_finger">
    <visual>
      <geometry>
        <box size="0.055 0.005 0.01"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="0.0275 0 0"/>
    </visual>
  </link>

  <link name="right_lower_finger">
    <visual>
      <geometry>
        <box size="0.055 0.005 0.01"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="0.0275 0 0"/>
    </visual>
  </link>

  <link name="right_palm">
    <visual>
      <geometry>
        <box size="0.01 0.005 0.05"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
  </link>

  <link name="right_upper_finger_top">
    <visual>
      <geometry>
        <box size="0.05 0.005 0.01"/>
      </geometry>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="-0.025 0 0"/>
    </visual>
  </link>

  <link name="right_lower_finger_top">
    <visual>
      <geometry>
        <box size="0.05 0.005 0.01"/>
      </geometry>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="-0.025 0 0"/>
    </visual>
  </link>

  <link name="left_finger_top">
    <visual>
      <geometry>
        <box size="0.05 0.005 0.01"/>
      </geometry>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="0.025 0 0"/>
    </visual>
  </link>

  <link name="grab_object">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.02"/>
      </geometry>
      <material name="green">
        <color rgba="0.5 0.5 0 1"/>
      </material>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
  </link>

  <joint name="base_to_grab_object" type="prismatic">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="0.045" upper="0.4" velocity="0.5"/>
    <parent link="base_link"/>
    <child link="grab_object"/>
    <origin rpy="0 0 0" xyz="0 0.4 0"/>
  </joint>

  <joint name="base_to_left_finger" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-0.785" upper="0.0" velocity="0.5"/>
    <parent link="base_link"/>
    <child link="left_finger"/>
    <origin rpy="0 0 2.35" xyz="-0.0225 0 0"/>
  </joint>

  <joint name="base_to_right_palm" type="fixed">
    <parent link="base_link"/>
    <child link="right_palm"/>
    <origin rpy="0 0 0" xyz="0.0225 0 0"/>
  </joint>

  <joint name="right_palm_to_right_upper_finger" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0.0" upper="0.785" velocity="0.5"/>
    <parent link="right_palm"/>
    <child link="right_upper_finger"/>
    <origin rpy="0 0 0.785" xyz="0 0 0.025"/>
  </joint>

  <joint name="right_palm_to_right_lower_finger" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0.0" upper="0.785" velocity="0.5"/>
    <parent link="right_palm"/>
    <child link="right_lower_finger"/>
    <origin rpy="0 0 0.785" xyz="0 0 -0.025"/>
  </joint>

  <joint name="right_lower_finger_to_right_lower_finger_top" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0.0" upper="1.04" velocity="0.5"/>
    <parent link="right_lower_finger"/>
    <child link="right_lower_finger_top"/>
    <origin rpy="0 0 -2.617" xyz="0.055 0 0"/>
  </joint>

  <joint name="right_upper_finger_to_right_upper_finger_top" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0.0" upper="1.04" velocity="0.5"/>
    <parent link="right_upper_finger"/>
    <child link="right_upper_finger_top"/>
    <origin rpy="0 0 -2.617" xyz="0.055 0 0"/>
  </joint>

  <joint name="left_finger_to_left_finger_top" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.04" upper="0" velocity="0.5"/>
    <parent link="left_finger"/>
    <child link="left_finger_top"/>
    <origin rpy="0 0 5.759" xyz="0.055 0 0"/>
  </joint>
</robot>

